types/gatherer.d.ts

```diff
@@ -51,6 +51,21 @@ declare global {
       trace?: Trace;
     }
 
+    type PhaseResult_ = void|LH.GathererArtifacts[keyof LH.GathererArtifacts]
+    export type PhaseResult = PhaseResult_ | Promise<PhaseResult_>
+
+    export interface GathererInstance {
```

Comment:
same here. The js class is the base class and the interface. Duplicating it here doesn't seem to do more with the type and adds to maintenance/distancing types from jsdocs, etc

Reply:
Decoupling the implementation and the interface is the *goal* here to distinguish the new from the old and differentiate which gatherers support which models of execution in a relatively trivial and lightweight way.

Is there a counter proposal in there I'm missing for how incremental transition for gatherers will be supported? I'm happy to move the docs to the interface if that's what you're asking.